src/main/java/com/ita/if103java/ims/security/UserDetailsServiceImpl.java

```diff
@@ -1,26 +1,31 @@
 package com.ita.if103java.ims.security;
 
+import com.ita.if103java.ims.dao.AccountTypeDao;
 import com.ita.if103java.ims.dao.UserDao;
+import com.ita.if103java.ims.entity.AccountType;
+import com.ita.if103java.ims.entity.User;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.core.userdetails.UsernameNotFoundException;
 import org.springframework.stereotype.Service;
 
-import java.util.Optional;
-
 @Service
 public class UserDetailsServiceImpl implements UserDetailsService {
 
-    @Autowired
     private UserDao userDao;
+    private AccountTypeDao accountTypeDao;
+
+    @Autowired
+    public UserDetailsServiceImpl(UserDao userDao, AccountTypeDao accountTypeDao) {
+        this.userDao = userDao;
+        this.accountTypeDao = accountTypeDao;
+    }
 
     @Override
-    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
-        return new UserDetailsImpl(Optional.ofNullable(userDao.findByEmail(username))
-            .orElseThrow(() -> {
-                throw new UsernameNotFoundException("Error during obtain user with this username");
-            }));
+    public UserDetailsImpl loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userDao.findByEmail(username);
+        AccountType type = accountTypeDao.findById(user.getAccountId());
```

Comment:
I think here is a mistake. First of all we need to find Account by account ID from User and then find AccountType by accountTypeId from Account.
@DashaBoichuk Please check and cooperate with Maria on this.

Reply:
@MarrMika AccountType type = accountTypeDao.findById(accountDao.findById(user.getAccountId()).getTypeId());

Reply:
thanks)